app/src/views/RootLayout/index.tsx

```diff
@@ -66,17 +71,70 @@ export function Component() {
 
     const [languagePending, setLanguagePending] = useState(false);
 
-    // FIXME: To be made functional after the implications of cookie rejections are finalized
     const [
         isCookiesBannerVisible,
         { setFalse: hideCookiesBanner },
-    ] = useBooleanState(false);
+    ] = useBooleanState(true);
+
+    const loadGoogleAnalytics = useCallback(() => {
+        const consent = JSON.parse(sessionStorage.getItem('cookie-consent') || '{}');
+
+        if (isAuthenticated || consent.analytics) {
+            const win = window as CustomWindow;
+
+            if (!win.gtag) {
+                const script = document.createElement('script');
+                script.src = 'https://www.googletagmanager.com/gtag/js?id=UA-XXXXXXX-X';
+                script.async = true;
+                document.head.appendChild(script);
+
+                win.dataLayer = win.dataLayer || [];
+                win.gtag = (...args) => win.dataLayer!.push(args);
+            }
+
+            win.gtag('js', new Date());
+            // TODO Add Google Analytics ID
+            win.gtag('config', 'UA-XXXXXXX-X', { anonymize_ip: true });
+        }
+    }, [isAuthenticated]);
+
+    const handleAccept = useCallback(() => {
+        const consent = { analytics: true };
+        sessionStorage.setItem('cookie-consent', JSON.stringify(consent));
+
+        loadGoogleAnalytics();
+
+        hideCookiesBanner();
+    }, [
+        hideCookiesBanner,
+        loadGoogleAnalytics,
+    ]);
+
+    const handleReject = useCallback(() => {
+        sessionStorage.setItem('cookie-consent', JSON.stringify({ analytics: false }));
+
+        const gaScript = document.querySelector('script[src*="googletagmanager.com/gtag/js"]');
+        if (gaScript) {
+            gaScript.remove();
+        }
```

Comment:
we don't need to remove the script all-together. we should just update the consent.